### DISCUSSION TITLE
Does the G7 return values greater than 400

### DISCUSSION THREAD

**Discussion by bassettb** [NightscoutFoundation/xDrip#3826] [Q&A]
It seems that the G7 does support values over 400.
We use xDrip, and xDrip uploads the data to our Nightscout instance.  xDrip and Nightscout both simply show the word "HIGH" when BG exceeds 400 mg/dL.  

We also have a [SugarPiDisplay](https://github.com/bassettb/SugarPiDisplay) in our kitchen.  Since we started using the G7, our SugarPiDisplay now shows values over 400.  This python app is pulling values from Nightscout using the `/api/v1/entries/sgv` API.

So as far as I can tell, the G7 sends values over 400 (we've seen up to 700).  Which means that xDrip and Nightscout both store higher values.  
Can anyone confirm this?  And if it's true, could we update xDrip and Nightscout to somehow show higher numerical values?

**Comment by Navid200** [comment]
This is what the Dexcom G7 manual shows:  
<br/>  
![Screenshot 2024-12-23 000528](https://github.com/user-attachments/assets/b8bedc5b-930c-4b59-adde-f8215569dab3)  
<br/>  


If it happens next time when your glucose is that high, can you switch to Dexcom app and show us that indeed G7 provides readings greater than 400 by showing a screenshot of your Dexcom app?

**Reply by bassettb** [reply]
We don't have the Dexcom app installed on his phone.  
I downloaded the xDrip database from his phone this morning.  Here's a snippet from last night.
```
select br.calculated_value, br.source_info from BgReadings br 
where timestamp < 1734937977271
order by br."timestamp" desc limit 35;`
```

```
calculated_value|source_info|
----------------+-----------+
           320.0|G7         |
           331.0|G7         |
           320.0|G7         |
           312.0|G7         |
           339.0|G7         |
           390.0|G7         |
           418.0|G7         |
           420.0|G7         |
           434.0|G7         |
           427.0|G7         |
           431.0|G7         |
           421.0|G7         |
           424.0|G7         |
           451.0|G7         |
           458.0|G7         |
           450.0|G7         |
           448.0|G7         |
           456.0|G7         |
           476.0|G7         |
           495.0|G7         |
           488.0|G7         |
           497.0|G7         |
           543.0|G7         |
           561.0|G7         |
           563.0|G7         |
           564.0|G7         |
           569.0|G7         |
           588.0|G7         |
           598.0|G7         |
           587.0|G7         |
           608.0|G7         |
           602.0|G7         |
           579.0|G7         |
           551.0|G7         |
           542.0|G7         |
```

**Reply by Navid200** [reply]
How exactly did you get the database?  Did you do a save database and used the export file that xDrip created? 

I would like to do the same on my phone to see if I can figure out what is going on.

**Reply by bassettb** [reply]
Three dot menu -> Import/Export Features -> Export database

**Reply by Navid200** [reply]
I see the calculated value in agreement with what xDrip shows under data table on the right right before the arrow symbol.
Can you enable data tables and see if you see the same values greater than 400 in it?  You can scroll through the data table to see previous readings.

https://navid200.github.io/xDrip/docs/Datatables.html

**Reply by bassettb** [reply]
yes, the values on the right match what is in the database
![Screenshot_20241223_115657_xDrip+](https://github.com/user-attachments/assets/92cdf647-b506-4356-9d39-c7af6797062c)

**Reply by Navid200** [reply]
Would you please post a screenshot of the classic status page?

**Comment by bassettb** [comment]
![395348845392378783](https://github.com/user-attachments/assets/e100a132-0b1d-4965-9140-6cfec936f4ac)

**Reply by Navid200** [reply]
I am puzzled by this.  I did not expect a G7 to transmit values greater than 400.

We can change xDrip to show values greater than 400.  In fact some Libre sensors provide values greater.  But, we decided to limit following Dexcom's approach. 
I'm not sure if it will be a good idea for xDrip to show values the manufacturer do not consider to be valid.  
I don't believe the sensor accuracy has been verified for such high values.  I suspect the reason Dexcom specifies the range to have a max of 400 is that they have verified that range only.

I am curious if Clarity would show values greater than 400 and use those larger values in statistics.

Do you think it is a crucial information to have how much higher than 400 one is?  
If someone is higher than 400, they will need to take insulin and consider their diet and exercise in order to lower the glucose.  But, I don't believe it will be a good idea to scale the amount of insulin by how much higher glucose is.  Too much insulin can be deadly.  So, I am not sure there is any value lost by not showing values greater than 400.

Frankly, I don't understand why a G7 would send those values considering the manual clearly states that 400 is max.

**Comment by bassettb** [comment]
Those are all good points.  I completely agree with the concerns.  But I think there's some value in providing an indication of just how far over 400 you are.  A patient should pay even closer attention to 650 than they would 410.  
Maybe xDrip could show 400+, 450+, 500+, 600+, etc.  

I'm just glad we started a discussion about the G7 behavior.  Let's continue the conversation and see what the community wants.

**Reply by Navid200** [reply]
@jamorham 

This is my concern: According to Dexcom's user guide, values greater than 400 are not valid.  If we show them, we will be showing values that Dexcom does not approve.  As far as what xDrip is supposed to do, or not to do, this would worry me.  

I expect that there is an acceptable level of error.  I expect that Dexcom only sells devices that meet their specifications.  But, the error level that they guarantee is only guaranteed over the range they present in the specifications (40-400).  This means that for all we know, values shown above 400 could be significantly in error.

Even if the users want this, this is something that should be requested from Dexcom, not from xDrip.

**Comment by bassettb** [comment]
That sounds like the right approach.  Very glad this software is approached so thoughtfully .
